app/controllers/api/databases.php

```diff
@@ -368,6 +369,65 @@ function updateAttribute(
     return $attribute;
 }
 
+/**
+ * Recursively processes a collection to get it's relationships
+ *
+ * Returns true if the user has access to the collection
+ */
+function getCollectionRelationships(Document $collection, bool $mode, Database $dbForProject, int $depth, array &$collections): bool
```

Comment:
Can we set a resource and inject it instead of using a global function?

Reply:
@eldadfux, considering your comment over https://github.com/appwrite/appwrite/pull/5587#discussion_r1253434658, do you not want us to set any global functions as a resource? Some other cases where we have functions we want to re-use between endpoints are:

1. create user
2. [createAttribute](https://github.com/appwrite/appwrite/blob/9f2bf844c7acc4932ca377941061665cd9b96645/app/controllers/api/databases.php#L62)
3. [updateAttribute](https://github.com/appwrite/appwrite/blob/9f2bf844c7acc4932ca377941061665cd9b96645/app/controllers/api/databases.php#L212)
4. [avatarCallback](https://github.com/appwrite/appwrite/blob/9f2bf844c7acc4932ca377941061665cd9b96645/app/controllers/api/avatars.php#L19)
5. some graphql methods

Reply:
There are a few places where we compromised on this as we didn't had any other reasonable choice. Me and @lohanidamodar are working on a longer term solution to define routes as classes. This will allow us to better reuse or extend routes logic.